library/core/src/num/mod.rs

```diff
@@ -810,6 +813,27 @@ macro_rules! doit {
         fn checked_add(&self, other: u32) -> Option<Self> {
             Self::checked_add(*self, other as Self)
         }
+        #[inline]
+        unsafe fn unchecked_mul(&self, other: u32) -> Self {
+            // SAFETY: Unsafe
```

Comment:
I think you're trying to say "this has the same safety conditions as the inner function, but the current comment doesn't explain that very well.
```suggestion
            // SAFETY: Conditions of `Self::unchecked_mul` must be upheld by the caller.
```
Same for the rest of the safety comments.